### PR TITLE
Replace tarantoolctl with tt (CE part)

### LINKED ISSUE #3501: Deprecate tarantoolctl and cartridge-cli in docs
Original request:
"Мы начинаем рекомендовать нашим пользователям использовать tt на замену старым CLI инструментам.

При этом в документации осталось немало гайдов, где показано как сделать те или иные операции с помощью старых cli.

Наша задача актуализировать эти гайды и заменить все на команды tt cli.
Изменения необходимо выполнить в Tarantool 2.11, Tarantool EE 2.11.
Разделы относящиеся к tarantoolctl и cartridge-cli также пометить как deprecated и рекомендовать переходить на tt cli"

## Description

[tt CLI](https://github.com/tarantool/tt/) is released and ready to replace `tarantoolctl` and `cartridge-cli`. We need to adjust command-line instruction across Tarantool docs to use the new tool.

`tt` must be the recommended Tarantool CLI since version 2.11. Thus, we will replace `tarantoolctl` and `cartridge-cli` in Tarantool CE and Tarantool EE **2.11** docs. 

`tarantoolctl` and `cartridge-cli` must be marked as deprecated tools. Additionally, we should provide instructions on migration from them to `tt`. They are available in the [tt repo](https://github.com/tarantool/tt/blob/master/doc/examples.md#transition-from-tarantoolctl-to-tt)

The scope of work is defined in [Notion](https://www.notion.so/tarantool/Tarantool-3-0-78ac037bcfbd4b068d0b409c06b3749e). It includes:
- CE docs:
  - 30-40 examples of tarantoolctl usage that require writing a tt alternative.
  - 80 occurrences of the word `tarantoolctl` that can be trivially replaced with ``tt``.
  - 20 other occurrences
- EE docs:
  -  11 occurrences of the word `tarantoolctl` that can be trivially replaced with ``tt``.

## Pages affected the most

- **Administration** section
  - https://www.tarantool.io/en/doc/latest/book/admin/start_stop_instance/
  - https://www.tarantool.io/en/doc/latest/book/admin/daemon_supervision/
  - https://www.tarantool.io/en/doc/latest/book/admin/instance_config/
  - https://www.tarantool.io/en/doc/latest/book/admin/logs/
  - https://www.tarantool.io/en/doc/latest/book/admin/server_introspection/
  - https://www.tarantool.io/en/doc/latest/book/admin/troubleshoot/
- How-to
  - https://www.tarantool.io/en/doc/latest/how-to/vshard_quick/
- EE docs: 
  - https://www.tarantool.io/en/enterprise_doc/dev/

## Question

Q: Do we need to replace `cartridge-cli` calls in **Cartridge docs** with `tt`?

A: No, a deprecation note with a link to tt docs is enough for Cartridge docs.

## Definition of done

- [x] All command-line calls in Tarantool CE and EE docs use tt
- [x] tarantoolctl reference page contains a deprecation note and a link to tt as a replacement
- [x] cartridge docs contain a deprecation note for cartridge-cli (but still use it in cartridge-specific examples)
- [x] Tarantool CE docs provide guidelines for migration from tarantoolctl and cartridge-cli to tt

### PR DESCRIPTION
**Pull request by p7nov**
Resolves #3501 

Deployment: https://docs.d.tarantool.io/ru/doc/gh-3501-tarantoolctl-to-tt/

Affected pages (to check specific changes on staging):
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/daemon_supervision/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/disaster_recovery/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/instance_config/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/logs/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/modules/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/os_notes/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/replication/repl_recover/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/security/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/server_introspection/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/troubleshoot/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/book/admin/vshard_admin/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/concepts/data_model/migrations/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/dev_guide/internals/iproto/requests/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/how-to/app/creating_app/avro_schemas/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/how-to/sql/improving_mysql/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/how-to/vshard_quick/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/configuration/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/reference_lua/fiber/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/tooling/interactive_console/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/tooling/tarantoolctl/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/tooling/tt_cli/
- https://docs.d.tarantool.io/en/doc/gh-3501-tarantoolctl-to-tt/reference/tooling/tt_cli/configuration/

### CODE REVIEW

File: doc/book/admin/logs.rst [hunk 1]
```diff
@@ -56,21 +54,21 @@ this file is usually like:
...
+:ref:`tt logrotate <tt-logrotate>` command to request instances to reopen their log
 files after they were moved by the program of your choice.
 
 Tarantool can write its logs to a log file, ``syslog`` or a program specified
 in the configuration file (see :ref:`log <cfg_logging-log>` parameter).
 
-By default, logs are written to a file as defined in ``tarantoolctl``
-defaults. ``tarantoolctl`` automatically detects if an instance is using
+By default, logs are written to a file as defined in ``tt``
+defaults. ``tt`` automatically detects if an instance is using
```
Comment (psergee) [inline doc/book/admin/logs.rst]: As I know, `tt` does not detect instance logging method. I'm not sure if `tarantoolctl` is able to do it. @LeonidVas do you know something about this functionality?

File: doc/concepts/data_model/migrations.rst [hunk 2]
```diff
@@ -113,14 +113,14 @@ This method is described in the README file of the
     There are also two other methods that we **do not recommend**,
     but you may find them useful for one reason or another.
 
-    **Method 3**: the ``tarantoolctl`` utility
+    **Method 3**: the :ref:`tt <tt-cli>` utility
 
-    The ``tarantoolctl`` utility ships with Tarantool.
-    Connect to the necessary instance using ``tarantoolctl connect``.
+    The ``tt`` utility ships with Tarantool.
```
Comment (psergee) [inline doc/concepts/data_model/migrations.rst]: As I know, it does not. There is a separate package for `tt`, which is a recommended dependency of the `tarantool` package.

File: doc/reference/tooling/tt_cli/configuration.rst [hunk 3]
```diff
@@ -69,6 +71,13 @@ app section
     Default: ``var/lib``.
 *   ``vinyl_dir`` -- the directory where vinyl files or subdirectories are stored.
     Default: ``var/lib``.
+
+    .. note::
+
+        In all directories specified in ``*_dir`` parameters, ``tt`` creates separate
+        directories for each instance in the environment. Their names match the instances'
```
Comment (psergee) [inline doc/reference/tooling/tt_cli/configuration.rst]: Not sure if this is important. Separate directories for applications. Each app directory has separate sub-dirs for instances.

File: doc/concepts/data_model/migrations.rst (hunk 2 above)
Comment (p7nov) [reply doc/concepts/data_model/migrations.rst]: Fixed.

File: doc/reference/tooling/tt_cli/configuration.rst (hunk 3 above)
Comment (p7nov) [reply doc/reference/tooling/tt_cli/configuration.rst]: Clarified this in the text.

File: doc/reference/tooling/tt_cli/configuration.rst [hunk 4]
```diff
@@ -69,6 +71,13 @@ app section
     Default: ``var/lib``.
 *   ``vinyl_dir`` -- the directory where vinyl files or subdirectories are stored.
     Default: ``var/lib``.
+
+    .. note::
+
+        In all directories specified in ``*_dir`` parameters, ``tt`` creates a
+        directory for each application and instance directories inside it.
+        Names is these directories match the names of applications and instances.
```
Comment (psergee) [inline doc/reference/tooling/tt_cli/configuration.rst]: Did you mean "Names of these directories..."?

Comment (p7nov) [reply doc/reference/tooling/tt_cli/configuration.rst]: Sure, thanks

Comment (psergee) [reply doc/reference/tooling/tt_cli/configuration.rst]: Looks like, not pushed yet.

File: doc/book/admin/logs.rst (hunk 1 above)
Comment (p7nov) [reply doc/book/admin/logs.rst]: I've rechecked this page. Seems like it needs a bigger update. Will rewrite it to reflect the ``tt`` usage and behavior.

Comment (p7nov) [reply doc/book/admin/logs.rst]: Rewrote the logs page, please check.

**Review by psergee — APPROVED** [review]

File: doc/book/admin/os_notes.rst [hunk 5]
```diff
@@ -10,7 +10,7 @@ Notes for operating systems
 Mac OS
 --------------------------------------------------------------------------------
 
-On Mac OS, you can administer Tarantool instances only with ``tarantoolctl``.
+On Mac OS, you can administer Tarantool instances only with :ref:`tt <tt-cli>`.
```
Comment (andreyaksenov) [inline doc/book/admin/os_notes.rst]:
```suggestion
On macOS, you can administer Tarantool instances only with :ref:`tt <tt-cli>`.
```

File: doc/book/admin/os_notes.rst [hunk 6]
```diff
@@ -10,7 +10,7 @@ Notes for operating systems
 Mac OS
```
Comment (andreyaksenov) [inline doc/book/admin/os_notes.rst]:
```suggestion
macOS
```

File: doc/reference/tooling/tarantoolctl.rst [hunk 7]
```diff
@@ -1,8 +1,13 @@
 .. _tarantoolctl:
 
---------------------------------------------------------------------------------
-Utility `tarantoolctl`
---------------------------------------------------------------------------------
+Utility tarantoolctl (deprecated)
```
Comment (andreyaksenov) [inline doc/reference/tooling/tarantoolctl.rst:3]: Maybe, rename to `tarantoolctl utility (deprecated)` for consistency with a neighboring `tt CLI utility`?

File: doc/reference/tooling/tt_cli/logrotate.rst [hunk 8]
```diff
@@ -1,3 +1,5 @@
+.. _tt-logrotate:
+
 Rotate instance logs
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/logrotate.rst]:
```suggestion
Rotating instance logs
```

File: doc/reference/tooling/tarantoolctl.rst [hunk 9]
```diff
@@ -214,3 +215,266 @@ Supported options:
...
+        log       = "./log/tarantool",
+        language  = "Lua",
+    }
+    instance_dir = "./instances.enabled"
+
+    $ tt init
+    • Found existing config '.tarantoolctl'
+    • Environment config is written to 'tt.yaml'
+
+After that, you can start manage Tarantool instances in this environment with ``tt``:
```
Comment (xuniq) [inline doc/reference/tooling/tarantoolctl.rst]:
```suggestion
After that, you can start managing Tarantool instances in this environment with ``tt``:
```

File: doc/reference/tooling/tt_cli/external_modules.rst [hunk 10]
```diff
@@ -1,3 +1,5 @@
+.. _tt-external_modules::
```
Comment (xuniq) [inline doc/reference/tooling/tt_cli/external_modules.rst]:
```suggestion
.. _tt-external_modules:
```

File: doc/book/admin/index.rst [hunk 11]
```diff
@@ -3,18 +3,16 @@
...
 
 Tarantool is designed to have multiple running instances on the same host.
 
 Here we show how to administer Tarantool instances using any of the following
 utilities:
 
 * ``systemd`` native utilities, or
-* :ref:`tarantoolctl <tarantoolctl>`, an administrative utility shipped and installed as
-  part of Tarantool distribution.
+* :ref:`tt <tt-cli>`, an command-line utility for managing Tarantool-based applications.
```
Comment (xuniq) [inline doc/book/admin/index.rst]:
```suggestion
* :ref:`tt <tt-cli>`, a command-line utility for managing Tarantool-based applications.
```

File: doc/reference/tooling/tt_cli/index.rst [hunk 12]
```diff
@@ -26,4 +22,40 @@ concept explanation, and the command reference.
...
     commands
     external_modules
-    multiple_instances
\ No newline at end of file
+    multiple_instances
+
+tt environments
+---------------
+
+The key aspect of the ``tt`` usage is an *environment*. An ``tt`` environment
```
Comment (xuniq) [inline doc/reference/tooling/tt_cli/index.rst]:
```suggestion
The key aspect of the ``tt`` usage is an *environment*. A ``tt`` environment
```

File: doc/book/admin/os_notes.rst (hunk 5 above)
Comment (xuniq) [inline doc/book/admin/os_notes.rst]: Section FreeBSD, lines 22, 32, 28-30:

1) tarantoolctl or tt?
2) improve bullet list:
* ``default`` for ``tarantoolctl`` defaults (see example below)
* ``instances.available`` for all available instance files
* ``instances.enabled`` for instance files to be auto-started by sysvinit

File: doc/reference/tooling/tarantoolctl.rst [hunk 13]
```diff
@@ -1,8 +1,13 @@
...
-Utility `tarantoolctl`
---------------------------------------------------------------------------------
+Utility tarantoolctl (deprecated)
+=================================
+
+.. important::
+
+    ``tarantoolctl`` is deprecated in favor of :ref:`tt CLI <tt-cli>`.
+    Find the instructions on switching from ``tarantoolctl`` to ``tt`` in
+    :ref:`Migration from tarantoolctl to tt <tarantoolctl-migration-to-tt>`.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tarantoolctl.rst:10]: > See also ``tarantoolctl`` usage examples in :ref:`Server administration <admin>`
section.

Am I correct that there are no more examples for tarantoolctl and this sentence can be removed?

File: doc/reference/tooling/tarantoolctl.rst [hunk 14]
```diff
@@ -214,3 +215,266 @@ Supported options:
...
+.. _tarantoolctl-config_file:
+
+tarantoolctl configuration file
+-------------------------------
+
+The ``tarantoolctl`` configuration file named ``.tarantoolctl``
+contains the configuration that ``tarantoolctl`` uses to
+override instance configuration. In other words, it contains system-wide
+configuration defaults. If ``tarantoolctl`` fails to find this file with
+the method described in section
```
Comment (andreyaksenov) [inline doc/reference/tooling/tarantoolctl.rst]:
```suggestion
the method described in the section
```

File: doc/reference/tooling/tarantoolctl.rst [hunk 15]
```diff
@@ -214,3 +215,266 @@ Supported options:
...
+  | Directory for snapshot .snap files; ``tarantoolctl`` will add
+    "/instance_name" to the directory name.
+* | ``vinyl_dir``
+  | Directory for vinyl files; ``tarantoolctl`` will add "/instance_name" to the
+    directory name.
+* | ``log``
+  | The place where the application log will go; ``tarantoolctl`` will add
+    "/instance_name.log" to the name.
+* | ``username``
+  | The user that runs the Tarantool instance. This is the operating-system user
```
Comment (andreyaksenov) [inline doc/reference/tooling/tarantoolctl.rst]:
```suggestion
  | The user that runs the Tarantool instance. This is the operating system user
```

File: doc/reference/tooling/tarantoolctl.rst [hunk 16]
```diff
@@ -214,3 +215,266 @@ Supported options:
...
+    ○ tarantool@example.service - Tarantool Database Server
+        Loaded: loaded (/lib/systemd/system/tarantool@.service; enabled; vendor preset: enabled)
+        Active: inactive (dead)
+
+Local configuration
+~~~~~~~~~~~~~~~~~~~
+
+If you have a local ``tarantoolctl`` configuration, create a ``tt`` environment
+based on the existing ``.tarantoolctl`` configuration file. To do this, run
+``tt init`` in the directory where the file in located.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tarantoolctl.rst]:
```suggestion
``tt init`` in the directory where the file is located.
```

File: doc/reference/tooling/tt_cli/index.rst [hunk 17]
```diff
@@ -7,15 +7,11 @@ tt CLI utility
 Tarantool-based applications. It covers a wide range of tasks -- from installing
 a specific Tarantool version to managing remote instances and developing applications.
 
-A multi-purpose tool for working with Tarantool from the command line, ``tt`` is
-a potential replacement for :ref:`tarantoolctl <tarantoolctl>`
-and :doc:`Cartridge CLI </book/cartridge/cartridge_cli/index>`.
-
 ``tt`` is developed in its own `GitHub repository <https://github.com/tarantool/tt>`_.
 Here you can find its source code, the changelog, and usage examples.
```
Comment (andreyaksenov) [inline doc/reference/tooling/tt_cli/index.rst]: > ... , and usage examples.

Given that documentation also contains usage examples (and, I believe, documentation is the main source of examples), this reference confuses me a bit.

**Review by andreyaksenov — APPROVED** [review]
LGTM, several minor fixes and suggestions

File: doc/book/admin/os_notes.rst (hunk 5 above)
Comment (p7nov) [reply doc/book/admin/os_notes.rst]: Moved this part to tarantooctl page; no FreeBSD specifics for tt (at least for now).

File: doc/reference/tooling/tt_cli/index.rst (hunk 17 above)
Comment (p7nov) [reply doc/reference/tooling/tt_cli/index.rst]: Rephrased.

**Pull request merged**